src/Semantics.js

```diff
@@ -196,6 +196,41 @@ Semantics.prototype.checkActionDicts = function() {
   }
 };
 
+Semantics.prototype.toRecipe = function() {
+  var sb = new common.StringBuffer();
```

Comment:
Yeah, IIRC that's the kind of difference in performance that Pat saw when he measured both approaches. The funny thing is that when I started implementing Ohm, I went with the `StringBuffer` approach b/c I did a comparison in jsperf and found out that it was significantly faster than using `+=`. But that was then, and this is now.

This is one of those cases where it would be really nice to have PL support for separating meanings from optimizations...